tests/bluetooth/tester/src/btp_csip.c

```diff
@@ -210,13 +236,39 @@ static uint8_t btp_csip_set_coordinator_lock(const void *cmd, uint16_t cmd_len,
 static uint8_t btp_csip_set_coordinator_release(const void *cmd, uint16_t cmd_len, void *rsp,
 					     uint16_t *rsp_len)
 {
+	const struct bt_csip_set_coordinator_set_member *members[ARRAY_SIZE(btp_csip_set_members)];
 	const struct btp_csip_set_coordinator_release_cmd *cp = cmd;
+	unsigned long member_count = 0;
 	int err;
 
 	LOG_DBG("");
 
-	err = bt_csip_set_coordinator_release(btp_csip_set_members, cp->count,
-					      &cur_csis_inst->info);
+	if (cp->addr_cnt != 0) {
+		/* TODO: add support for lock release procedure on subset of set members */
+		/* If no address is passed, we perform procedure on all members */
+		return BTP_STATUS_FAILED;
+	}
+
+	if (cur_csis_inst == NULL) {
+		LOG_ERR("No CISP instance available");
+		return BTP_STATUS_FAILED;
+	}
+
+	for (size_t i = 0; i < (size_t)ARRAY_SIZE(btp_csip_set_members); i++) {
+		if (btp_csip_set_members[i] == NULL) {
+			continue;
+		}
+
+		members[member_count++] = btp_csip_set_members[i];
+	}
+
+	if (member_count == 0) {
+		LOG_ERR("No set members available");
+
+		return BTP_STATUS_FAILED;
+	}
```

Comment:
This is identical to the lock procedure. Can this be moved to a function that can be used by those 2 functions?

Reply:
Moved to function available_members